code/lib/core-server/src/utils/StoryIndexGenerator.ts

```diff
@@ -118,10 +118,12 @@ export class StoryIndexGenerator {
       this.specifiers.map(async (specifier) => {
         const pathToSubIndex = {} as SpecifierStoriesCache;
 
-        const fullGlob = slash(
-          path.join(this.options.workingDir, specifier.directory, specifier.files)
+        const fullGlob = slash(path.join(specifier.directory, specifier.files));
+        const files = await glob(
+          fullGlob,
+          // Do not treat workingDir path as a glob pattern
```

Comment:
It means that with `cwd`, we're now treating only `path.join(specifier.directory, specifier.files)` as glob. In case the comment brings more confusion than value, I'm happy to remove it.

Reply:
OK, I see now.

Would I need to do the same here:
https://github.com/storybookjs/storybook/pull/22186

https://github.com/storybookjs/storybook/pull/22186/files

Reply:
I'd say yes, in case there are special chars in the path, they won't match without it.